### PR TITLE
AB#469292 parent with no children or children with no parent

### PR DESCRIPTION
**Pull request by abidkayanidefra**
[AB#469292](https://dev.azure.com/defragovuk/67a63f24-5d0d-4c35-9d23-885593eb7354/_workitems/edit/469292) parent with no children or children with no parent

### CODE REVIEW

**Review by tsnatos — APPROVED** [review]

File: src/EPR.SubsidiaryBulkUpload.Application.UnitTests/Services/RecordExtractionTests.cs [hunk 1]
```diff
@@ -56,10 +56,13 @@ public void ShouldIgnoreRecordsWhereThereAreNoParents()
...
         // Act
-        var parentAndSubsidiaries = extraction.ExtractParentsAndSubsidiaries(subsidiaries);
+        var parentAndSubsidiaries = extraction.ExtractParentsAndSubsidiaries(subsidiaries).ToArray();
 
         // Assert
-        parentAndSubsidiaries.Should().BeEmpty();
+        parentAndSubsidiaries.Should().HaveCount(3);
+        parentAndSubsidiaries[0].Parent.organisation_name.Should().BeEquivalentTo("orphan");
+        parentAndSubsidiaries[1].Parent.organisation_name.Should().BeEquivalentTo("orphan");
+        parentAndSubsidiaries[2].Parent.organisation_name.Should().BeEquivalentTo("orphan");
```
Comment (mikeymonster) [inline src/EPR.SubsidiaryBulkUpload.Application.UnitTests/Services/RecordExtractionTests.cs:65]: You could use `.Should().AllBeEquivalentTo("orphan")` here and below

File: src/EPR.SubsidiaryBulkUpload.Application/Services/BulkSubsidiaryProcessor.cs [hunk 2]
```diff
@@ -60,8 +60,8 @@ public async Task<int> Process(IEnumerable<CompaniesHouseCompany> subsidiaries,
             .Where(subAndLink => subAndLink.LinkModel != null);
 
         var companyHouseAPIProcessStatistics = await ProcessCompanyHouseAPI(newSubsidiariesToAdd_DataFromLocalStorageOrCH, userRequestModel);
-        var remainingToProcessPart2 = remainingToProcess.Except(companyHouseAPIProcessStatistics.CompaniesHouseAPIErrorListReported);
-        var remainingToProcessPart3 = remainingToProcessPart2.Except(companyHouseAPIProcessStatistics.NewAddedSubsidiaries).Except(companyHouseAPIProcessStatistics.DuplicateSubsidiaries).Except(companyHouseAPIProcessStatistics.NotAddedSubsidiaries);
+        var remainingToProcessAfterAPIChecks = remainingToProcess.Except(companyHouseAPIProcessStatistics.CompaniesHouseAPIErrorListReported);
```
Comment (mikeymonster) [inline src/EPR.SubsidiaryBulkUpload.Application/Services/BulkSubsidiaryProcessor.cs:63]: Finally! New names :)

**Review by tsnatos — DISMISSED** [review]

File: src/EPR.SubsidiaryBulkUpload.Application/Services/BulkSubsidiaryProcessor.cs (hunk 2 above)
Comment (abidkayanidefra) [reply src/EPR.SubsidiaryBulkUpload.Application/Services/BulkSubsidiaryProcessor.cs:63]: lol. yup. some refactoring after pressure released.

File: src/EPR.SubsidiaryBulkUpload.Application.UnitTests/Services/RecordExtractionTests.cs (hunk 1 above)
Comment (abidkayanidefra) [reply src/EPR.SubsidiaryBulkUpload.Application.UnitTests/Services/RecordExtractionTests.cs:65]: all three parents are not same. Id are different. only the name is used like that.

**Review by mikeymonster — APPROVED** [review]

**Review by Fahad-Ali-Eviden — APPROVED** [review]

**Pull request merged**